compiler-core/src/parse/tests.rs

```diff
@@ -1014,6 +1014,21 @@ type A {
     );
 }
 
+// Tests whether diagnostic presents an example of how to formulate a proper
+// record constructor based off a common user error pattern.
+// https://github.com/gleam-lang/gleam/issues/3324
+
+#[test]
+fn type_invalid_variant_constructor() {
+    assert_module_error!(
+        "
+pub type User {
+    name: String,
+}
+"
+    );
+}
```

Comment:
Could we have tests for each of the syntaxes that this feature supports please 🙏  I can see there's at least 4 variants